src/ng/http.js

```
@@ -1135,29 +1135,86 @@ function $HttpProvider() {
       }
       return url;
     }
   }];
 }
+
+/**
+ * @ngdoc provider
+ * @name $httpUrlParamsProvider
+ * @description
+ * Use `$httpUrlParamsProvider` to change the default behavior of the {@link ng.$httpUrlParams $httpUrlParams} service.
+ * */
+function $HttpUrlParamsProvider() {
+
+  var paramSerializers = createMap();
+
+  function serializeParams(params, addArrayMarker) {
+    var parts = [];
+
+    if (!params) return '';
+
+    forEachSorted(params, function(value, key) {
+      if (value === null || isUndefined(value)) return;
+      if (!isArray(value)) value = [value];
+
+      forEach(value, function(v) {
+        if (isObject(v)) {
+          v = isDate(v) ? v.toISOString() : toJson(v);
+        }
+        parts.push(encodeUriQuery(key) + (addArrayMarker ? '[]' : '') + '=' + encodeUriQuery(v));
+      });
+    });
+
+    return parts.join('&');
+  }
+
+
+  this.defaultMode = 'traditional';
+
+
+  this.registerSerializer = function registerSerializer(mode, serFn) {
+    paramSerializers[mode] = serFn;
+  };
+
+  /**
+  * @ngdoc service
+  * @name $httpUrlParams
+  *
+  * @description
+  * The `$httpUrlParams` service is responsible for serializing request params
+  * (expressed as a JavaScript object) to a string.
+  * It abstracts the way in which request params are transformed, grouped, encoded etc.
+  *
+  * Normally you wouldn't use this service directly in the application's code but rather override this service
+  * to enable custom serialization schemas for URL parameters.
+  *
+  * The `$httpUrlParams` service comes handy while unit testing with {@link ngMock.$httpBackend $httpBackend mock},
+  * as one can inject `$httpUrlParams` into a test and serialize URL params as {@link ng.$http $http} service.
+  */
+  this.$get = function() {
+
+    var provider = this;
+    var HttpUrlParams = {};
+
+    HttpUrlParams.serialize = function(params, mode) {
+      return paramSerializers[lowercase(mode) || provider.defaultMode](params);
```

Should we throw if a mode is not available? I think we should.